cherab/core/laser/node.pyx

```diff
@@ -0,0 +1,249 @@
+        """
+        self._importance = 0.
+        self._geometry = []
+
+    @property
+    def plasma(self):
+        return self._plasma
+
+    @plasma.setter
+    def plasma(self, Plasma value not None):
+
+        #unregister from old plasma notifier
+        if self._plasma is not None:
+            self._plasma.notifier.remove(self._plasma_changed)
+
+        self._plasma = value
+        self._configure()
+
+    @property
+    def importance(self):
+        return self._importance
+
+    @importance.setter
+    def importance(self, double value):
+        
+        self._importance = value
+        self._configure()
+
+    def _configure(self):
+        """
+        Reconfigure the laser primitives and update references after changes.
+        """
+
+        # no further work if there are no emission models
+        if not list(self._models):
+            return
+        
+        #no further work if there is no laser profile connected
+        if self._laser_profile is None:
+            return
+
+        # update the plasma reference in models
+        for model in list(self._models):
+            model.plasma = self._plasma
+
+        # clear geometry parents to remove segments
+        for i in self._geometry:
+            i.parent = None
+        self._geometry = []
+
+        # rebuild geometry
+        primitives = self._laser_profile.generate_geometry()
+
+        for i in primitives:
+            i.parent = self
+            i.material = LaserMaterial(self, i, list(self._models), self._integrator)
+
+        self._geometry = primitives
+
+    @property
+    def laser_spectrum(self):
+        return self._laser_spectrum
+
+    @laser_spectrum.setter
+    def laser_spectrum(self, LaserSpectrum value):
+        self._laser_spectrum = value
+        self._configure()
+
+    @property
+    def laser_profile(self):
+        return self._laser_profile
+
+    @laser_profile.setter
+    def laser_profile(self, LaserProfile value):
+
+        self._laser_profile = value
+        self._laser_profile.laser = self
+
+        self._configure()
+
+    @property
+    def models(self):
+        return list(self._models)
+
+    @models.setter
+    def models(self, value):
+        
+        # check necessary data is available
+        if not all([self._plasma, self._laser_profile, self._laser_spectrum]):
+            raise ValueError("The plasma, laser_profile and laser_spectrum must be set before before specifying any models.")
+
+        self._models.set(value)
+        self._configure()
+
+    @property
+    def integrator(self):
+        return self._integrator
+
+    @integrator.setter
+    def integrator(self, VolumeIntegrator value):
+        self._integrator = value
+        self._configure()
+
+    def get_geometry(self):
+        return self._geometry
+
+    def _plasma_changed(self):
+        """React to change of plasma and propagate the information."""
+        self._configure()
+
+    def _modified(self):
+        self._configure()
```

Comment:
`_modified` is defined [here](https://github.com/raysect/source/blob/3c1f2f2f1cce9c2a2d5ca2952a70af3287d53dbf/raysect/core/scenegraph/_nodebase.pyx#L174). Because there are transforms cached in material, it should be rebuilt when the `Laser` transform changes I think

Reply:
I see, thanks.